### ISSUE TITLE
How do you debug the project in order to contribute?

### ISSUE THREAD

**Issue by beeradmoore** [jamesmontemagno/MediaPlugin#641]
I want to fix some of these image resize/rotation issues for iOS and Android but I can't figure out how to even start contributing to this package. 

I've forked it and I'm working from the master branch. I first tried to add Media.Plugin.csproj to my Xamarin.Forms (.NET Standard 2.0) project which is how I've normally done things in the past. After I added it I was unable to restore its NuGet packages.

44 or so issues all very similar to this.
```
/Users/bradmoore/git/MediaPlugin/src/Media.Plugin/Media.Plugin.csproj: Error NU1202: Package Xamarin.Android.Support.Core.Utils 25.4.0.2 is not compatible with netstandard1.0 (.NETStandard,Version=v1.0). Package Xamarin.Android.Support.Core.Utils 25.4.0.2 supports: monoandroid70 (MonoAndroid,Version=v7.0) (NU1202) (Media.Plugin)
```

So I tried to make just a Xamarin.iOS project and I had very similar issues. I thought maybe I am meant to open Media.sln and build a dll in debug mode and add that to my project but I get the same 44 errors as above. I even tried Media.sln in the test folder and attempted to build MediaTest.iOS for iPhoneSimulator but still no dice. 

I'm out of ideas, how are you meant to debug this library?

**Comment by jamesmontemagno** [comment]
You would need to build it on VS 2017/2019 on Windows since I use multitargeting. You could build via the command line and call msbuild on mac and that would work too, but much more complicated.

**Comment by beeradmoore** [comment]
Ah k. I'll boot up my Windows VM later and see how I go.

**Comment by wegascee** [comment]
What finally worked for me in VS 2017:

1. Open *MediaSample.sln*
2. Added *Media.Plugin.csproj* from *src/Media.Plugin* folder and build it
3. Edited Info.plist to match provisioning profile
4. Uninstalled the NuGet package *Xam.Plugin.Media* and added reference to *Media.Plugin/Media.Plugin.csproj* (references finally *src\Media.Plugin\obj\Debug\netstandard1.0\Plugin.Media.dll*)
5. Set a breakpoint on *iOS/MediaImplementation.cs* `TakePhotoAsync()`

But currently I don't understand how the multitargeting works. The source project doesn't include the iOS folder in the solution, only some dependencies.

Regarding the rotation issue I found the following out:

1. If you use `AllowCropping = true` the image is rotated wrong (rotation to the right is overdone) on iOS.
2. If you use the front camera (default in the sample) on iOS the rotation left/right is switched. This is independent from `AllowCropping`.

I wasn't able to debug, because `RotateImage()` isn't called or setting the breakpoint doesn't work with the anonymous method.
